ziggurat-core-geoip/src/geoip.rs

```diff
@@ -0,0 +1,130 @@
+//use ipinfo::{IpInfo, IpInfoConfig};   // TODO(asmie): implement IpInfo provider
+use ip2location::{Record, DB};
+use ipgeolocate::{Locator, Service};
+
+/// List of supported geoip providers.
+pub enum Provider {
```

Comment:
Should we implement Copy trait here? since this is an exposed API
See this:
https://doc.rust-lang.org/stable/std/marker/trait.Copy.html#when-should-my-type-be-copy